src/callbacks.jl

```diff
@@ -41,7 +41,15 @@ function addheuristiccallback(m::Model, f::Function)
 end
 function addinfocallback(m::Model, f::Function; when::Symbol = :Undefined)
     if when == :Undefined
-        error("Info Callbacks without a `when` clause are unsupported.")
+        when = :Intermediate
```

Comment:
Doesn't really matter, but I think you could do something like:
```jl
function _not_specified()
    warn("""Info Callbacks without a when clause will currently default
              to fire only in the :Intermediate state to preserve its behavior
              with previous versions of JuMP.
 
             This behavior will be deprecated in subsequent versions of JuMP, so
             please rewrite all invocations of addinfocallbacks(m, f) to
             addinfocallbacks(m, f, when = :Intermediate) instead.
             """)
    :Intermediate
end

function addinfocallback(m::Model, f::Function; when::Symbol = _not_specified())
...
```
to avoid the corner case where I might pass in ``when=:Undefined``.